avcheck.py

```diff
@@ -0,0 +1,606 @@
+                        binary_path = service.get_binary_path()
+                        
+                        av_services.append({
+                            'name': service_name,
+                            'display': display_name,
+                            'state': state,
+                            'binary': binary_path
+                        })
+                        
+                except Exception as e:
+                    vollog.debug(f"Error processing service: {e}")
+                    continue
+                    
+        except Exception as e:
+            vollog.debug(f"Service scan unavailable: {e}")
+        
+        return av_services
+    
+    def _detect_av_dlls(self, kernel_module_name: str) -> List[Dict]:
+        """Detect AV-related DLLs loaded in processes."""
+        av_dlls = []
+        dll_patterns = [
+            'defender', 'kaspersky', 'avg', 'avira', 'bitdefender', 'mcafee',
+            'norton', 'symantec', 'eset', 'malwarebytes', 'sophos', 'trend',
+            'avast', 'comodo', 'fsecure', 'panda', 'gdata', 'webroot',
+            'scan', 'antivirus', 'av', 'guard', 'shield', 'protect'
+        ]
+        
+        seen_dlls = set()
+        
+        try:
+            for proc in pslist.PsList.list_processes(self.context, kernel_module_name):
+                try:
+                    proc_name = proc.ImageFileName.cast(
+                        "string",
+                        max_length=proc.ImageFileName.vol.count,
+                        errors="replace"
+                    )
+                    
+                    for dll in proc.load_order_modules():
+                        try:
+                            dll_name = dll.BaseDllName.get_string().lower()
+                            dll_path = dll.FullDllName.get_string()
+                            
+                            if any(pattern in dll_name for pattern in dll_patterns):
+                                dll_key = f"{dll_name}|{dll_path}"
+                                if dll_key not in seen_dlls:
+                                    seen_dlls.add(dll_key)
+                                    av_dlls.append({
+                                        'name': dll_name,
+                                        'path': dll_path,
+                                        'process': proc_name,
+                                        'pid': proc.UniqueProcessId,
+                                        'base': format_hints.Hex(dll.DllBase)
+                                    })
+                                    
+                        except Exception:
+                            continue
+                            
+                except Exception:
+                    continue
+                    
+        except Exception as e:
+            vollog.debug(f"Error scanning DLLs: {e}")
+        
+        return av_dlls
+    
+    def _detect_av_files(self) -> List[Dict]:

```

Comment:
Again, ideally this should take a `context` and a `kernel` module as parameters.